src/averageAges.js

```diff
@@ -55,7 +65,32 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const female = people.filter(person => person.sex === 'f');
+
+  const withChild = female.filter(woman => {
+    return people.filter(person => person.mother === woman.name);
+  });
+
+  const children = onlyWithSon
+    ? people.filter(person => withChild.some(mother => {
+      return mother.name === person.mother && person.sex === 'm';
+    }))
+    : people.filter(person => {
+      return withChild.some(mother => mother.name === person.mother);
+    });
+
+  let calc = 0;
+
+  const result = children.reduce(function(prev, item) {
```

Comment:
It's incorrect to name variable result and then not return it as a result of the function.